kernel/src/hil/sensors.rs

```diff
@@ -112,3 +112,26 @@ pub trait NineDofClient {
     /// over the syscall interface to an application.
     fn callback(&self, arg1: usize, arg2: usize, arg3: usize);
 }
+
+/// Basic Interface for Sound Pressure
+pub trait SoundPressure<'a> {
```

Comment:
Please add comments that explain the state machine. For example, what happens if you call `disable()` then `read_sound_pressure()`? What happens if you call `read_sound_pressure()` then `disable()`? Specify what the valid ReturnCode values are for each call and what they mean. 

The callback needs to include return a ReturnCode. Any API where there is a callback can have a situation where the initial request succeeds, but acquisition fails due to a later error. E.g., if this API is virtualized, it can be that issuing a read just puts the request on a queue (or waits for access to a bus), but then when it's pulled off the queue the request fails.

Reply:
> The callback needs to include return a ReturnCode. Any API where there is a callback can have a situation where the initial request succeeds, but acquisition fails due to a later error. E.g., if this API is virtualized, it can be that issuing a read just puts the request on a queue (or waits for access to a bus), but then when it's pulled off the queue the request fails.

I agree, but Temperature, Humidity and others do not have one. I will add it here, should I add a PR with the ReturnCode for the others?